<a href="https://colab.research.google.com/github/viandre7/Viviana_GarciaMonje_KC/blob/main/Taller_Semana_8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**1. TALLER SEMANA 8**



## **1.1 BREAST CANCER (CLASIFICASIÖN)**


### **1.1.1 Notación**

**1) ID number** (Se elimina)

**2) Diagnosis** (M = malignant, B = benign) 
Ten real-valued features are computed for each cell nucleus:

**a) radius** (mean of distances from center to points on the perimeter)

**b) texture** (standard deviation of gray-scale values)

**c) perimeter**

**d) area**

**e) smoothness** (local variation in radius lengths)

**f) compactness** (perimeter^2 / area - 1.0)

**g) concavity** (severity of concave portions of the contour)

**h) concave points** (number of concave portions of the contour)

**i) symmetry**

**j) fractal dimension** ("coastline approximation" - 1)

###**1.1.2 Conectarse al drive**


In [1]:
from google.colab import drive
import os
drive.mount('/content/gdrive')
# Establecer ruta de acceso en drive
import os
print(os.getcwd())
os.chdir("/content/gdrive/My Drive")

Mounted at /content/gdrive
/content


In [2]:
%cd '/content/gdrive/MyDrive/G2 - Data Sience 2021 (Keycode)/Taller_Semana_8'

/content/gdrive/MyDrive/G2 - Data Sience 2021 (Keycode)/Taller_Semana_8


###**1.1.3 Cargar paquetes**


In [3]:
#Carga de las librerías
import urllib.request
import pandas as pd

from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score



###**1.1.4 Cargar los datos - Breast_cancer (Clasificación)**


In [45]:
data_cancer= pd.read_csv('Breast_cancer_1.csv',sep=',')
data_cancer= data_cancer.drop(columns=['id'])
data_cancer.head()

diagnosis  radius_mean  ...  fractal_dimension_worst  Unnamed: 32
0         M        17.99  ...                  0.11890          NaN
1         M        20.57  ...                  0.08902          NaN
2         M        19.69  ...                  0.08758          NaN
3         M        11.42  ...                  0.17300          NaN
4         M        20.29  ...                  0.07678          NaN

[5 rows x 32 columns]

**Nota** 

Eliminamos la columna id, ya que no nos aporta en nada, sumado a que estos modelos no deben incluir dicha columna para su análisis

In [46]:
data_cancer.shape

(569, 32)

###**1.1.5 Remmplazar datos en diagnosis**

In [47]:
data_cancer['diagnosis1']=data_cancer.diagnosis.replace(to_replace=['B', 'M'], value=[0, 1])
data_cancer.head()

diagnosis  radius_mean  ...  Unnamed: 32  diagnosis1
0         M        17.99  ...          NaN           1
1         M        20.57  ...          NaN           1
2         M        19.69  ...          NaN           1
3         M        11.42  ...          NaN           1
4         M        20.29  ...          NaN           1

[5 rows x 33 columns]

In [48]:
data_cancer.drop(['diagnosis', 'Unnamed: 32'], axis = 'columns', inplace=True)

**Nota** 

Creamos una columna nueva. diagnosis1, para guardar los datos de la columna diagnosis que vamos a reemplazar (M,B) por 0 y 1, donde 0 es Benigno y 1 es Maligno, posteriormente eliminamos la colomna diagnosis ya que no la necesitamos, y la columna Unnamed pues no nos aporta nada.

In [49]:
data_cancer.head()

radius_mean  texture_mean  ...  fractal_dimension_worst  diagnosis1
0        17.99         10.38  ...                  0.11890           1
1        20.57         17.77  ...                  0.08902           1
2        19.69         21.25  ...                  0.08758           1
3        11.42         20.38  ...                  0.17300           1
4        20.29         14.34  ...                  0.07678           1

[5 rows x 31 columns]

In [50]:
data_cancer.isnull().sum()

radius_mean                0
texture_mean               0
perimeter_mean             0
area_mean                  0
smoothness_mean            0
compactness_mean           0
concavity_mean             0
concave points_mean        0
symmetry_mean              0
fractal_dimension_mean     0
radius_se                  0
texture_se                 0
perimeter_se               0
area_se                    0
smoothness_se              0
compactness_se             0
concavity_se               0
concave points_se          0
symmetry_se                0
fractal_dimension_se       0
radius_worst               0
texture_worst              0
perimeter_worst            0
area_worst                 0
smoothness_worst           0
compactness_worst          0
concavity_worst            0
concave points_worst       0
symmetry_worst             0
fractal_dimension_worst    0
diagnosis1                 0
dtype: int64

In [51]:
data_cancer.shape

(569, 31)

**NOTA**

El dataset se compone de 31 columnas de las cuales 30 corresponden a variables independientes, que usaremos para predecir.

diagnosis1 es la variable a predecir. Todas las variables son numéricas, ya sea enteras o reales y no tiene valores nulos.

###**1.1.6 Definir la matriz de diseño (X) y el vector respuesta (y)**

In [52]:
X = data_cancer.drop(["diagnosis1"], axis=1) # Tomar todas las columnas excepto la ultima
y = data_cancer["diagnosis1"] # Tomar el vector respuesta (ultima columna)

In [53]:
X

radius_mean  texture_mean  ...  symmetry_worst  fractal_dimension_worst
0          17.99         10.38  ...          0.4601                  0.11890
1          20.57         17.77  ...          0.2750                  0.08902
2          19.69         21.25  ...          0.3613                  0.08758
3          11.42         20.38  ...          0.6638                  0.17300
4          20.29         14.34  ...          0.2364                  0.07678
..           ...           ...  ...             ...                      ...
564        21.56         22.39  ...          0.2060                  0.07115
565        20.13         28.25  ...          0.2572                  0.06637
566        16.60         28.08  ...          0.2218                  0.07820
567        20.60         29.33  ...          0.4087                  0.12400
568         7.76         24.54  ...          0.2871                  0.07039

[569 rows x 30 columns]

In [54]:
y


0      1
1      1
2      1
3      1
4      1
      ..
564    1
565    1
566    1
567    1
568    0
Name: diagnosis1, Length: 569, dtype: int64

In [55]:
y.unique()

array([1, 0])

###**1.1.7 Separar en train y tests**

In [56]:
from sklearn.model_selection import train_test_split
(X_train, X_test,y_train, y_test) = train_test_split(X,y,stratify=y,test_size=0.30,random_state=42)

**Nota** Eligieremos el 70% de los datos para entrenamiento, y el 30% para test, de igual manera se le aplicara el random_state para seleccionar la semilla y los datos sean aleatoreos

###**1.1.8 Entrenar el modelo - Random Forest**

In [57]:
model = RandomForestClassifier(random_state=42, n_estimators=100,
class_weight="balanced", max_features="log2")
model.fit(X_train, y_train) #Se ajusta a los datos de entrenamiento

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight='balanced',
                       criterion='gini', max_depth=None, max_features='log2',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=100,
                       n_jobs=None, oob_score=False, random_state=42, verbose=0,
                       warm_start=False)

**Nota** 
Numero de estimadores 100 (arboles), tipo balanceado para que no queden muchas muestras separadas en la seleccion, tomara un log2 para tomar la cantidad de columnas para cada arbol

###**1.1.9 Sacar predicciones**

In [ ]:
# import numpy as np
# a=np.array(y_test)
# a

In [60]:
y_test_pred = model.predict(X_test) #Prediccion en Test
y_test_pred

array([0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1,
       1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 0,
       0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 1,
       1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0,
       0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1,
       0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0,
       0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1])

###**1.1.10 Calculamos el Acuracy**

In [61]:
# from sklearn.metrics import accuracy_score

test_accuracy = accuracy_score(y_test, y_test_pred)
print('% de aciertos sobre el set de evaluación:',test_accuracy)

% de aciertos sobre el set de evaluación: 0.9766081871345029


**Nota** obtenemos un acuracy de 0.97, es decir, una probabilidad de el 97% de los datos que predice sean correctos, de cada 100 datos que se clasifiquen el 97 quedaran bien clasificados, lo cual nos indica que es un muy buen modelo

###**1.1.11 Calculamos la precisión**

In [62]:
from sklearn.metrics import precision_score
precision_score(y_test, y_test_pred) 

1.0

**Nota**  El porcentaje de muestras que se clasifican bien dentro de cada clase, es aproximadamente del %

###**1.1.12 Matriz de confusion**

In [63]:
from sklearn.metrics import confusion_matrix
confusion_matrix(y_test, y_test_pred) 

array([[107,   0],
       [  4,  60]])

El modelo tuvo 107 verdaderos positivos, 0 falsos positivos, 60 verdaderos negativos y 4 falsos negativos, por lo que se puede concluir que el modelo funciona relativamente bien en cuanto a su clasificación

###**1.1.13 Calculamos el Recall**

In [64]:
from sklearn.metrics import recall_score
recall_score(y_test, y_test_pred) 

0.9375

**Nota** El modelo se adapta en un 93% a los datos al momento de clasificar

###**1.1.14 Calculamos f1 score**

In [65]:
from sklearn.metrics import f1_score
f1_score(y_test, y_test_pred) 

0.967741935483871

###**1.1.15 Conclusiones**

Todos los datos son similares, estan cerca, lo cual nos dice que el modelo es balanceado.

Con el random forest, obtenemos un acuracy de 0.97, es decir, el 97% de los datos que predice es correcto, lo cual nos indica que es un muy buen modelo

## **1.2 WINE (REGRESION)**


In [17]:
import pandas                  as pd
from   scipy import stats
import numpy                   as np
import matplotlib.pyplot       as plt
import seaborn                 as sns
import statsmodels.formula.api as sm
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

### **1.2.1 Notación**

1) Alcohol

2) Malic acid

3) Ash

4) Alcalinity of ash

5) Magnesium

6) Total phenols

7) Flavanoids

8) Nonflavanoid phenols

9) Proanthocyanins

10)Color intensity

11)Hue

12)OD280/OD315 of diluted wines

13)Proline

### **1.2.2 Cargar los datos**

In [5]:
data_wine= pd.read_csv('wine_1.csv',sep=',')
data_wine.head()

Wine  Alcohol  Malic.acid   Ash  ...  Color.int   Hue    OD  Proline
0     1    14.23        1.71  2.43  ...       5.64  1.04  3.92     1065
1     1    13.20        1.78  2.14  ...       4.38  1.05  3.40     1050
2     1    13.16        2.36  2.67  ...       5.68  1.03  3.17     1185
3     1    14.37        1.95  2.50  ...       7.80  0.86  3.45     1480
4     1    13.24        2.59  2.87  ...       4.32  1.04  2.93      735

[5 rows x 14 columns]

In [6]:
data_wine.shape

(178, 14)

### **1.2.3 Definir la matriz de diseño (X) y el vector respuesta (y)**

In [7]:
X=data_wine.iloc[:,:-1].values # Tomar todas las columnas excepto la ultima
y= data_wine.iloc[:,-1].values # Tomar el vector respuesta (ultima columna)

In [8]:
print(X.shape)
print(y.shape)

(178, 13)
(178,)


In [9]:
X

array([[ 1.  , 14.23,  1.71, ...,  5.64,  1.04,  3.92],
       [ 1.  , 13.2 ,  1.78, ...,  4.38,  1.05,  3.4 ],
       [ 1.  , 13.16,  2.36, ...,  5.68,  1.03,  3.17],
       ...,
       [ 3.  , 13.27,  4.28, ..., 10.2 ,  0.59,  1.56],
       [ 3.  , 13.17,  2.59, ...,  9.3 ,  0.6 ,  1.62],
       [ 3.  , 14.13,  4.1 , ...,  9.2 ,  0.61,  1.6 ]])

### **1.2.4 Normalizamos los datos, usando Standard scaler**

Las variables quedarán entre -2 y 2

In [10]:
from sklearn.preprocessing import StandardScaler
X_n= StandardScaler().fit_transform(X)
X_n

array([[-1.21394365,  1.51861254, -0.5622498 , ...,  0.25171685,
         0.36217728,  1.84791957],
       [-1.21394365,  0.24628963, -0.49941338, ..., -0.29332133,
         0.40605066,  1.1134493 ],
       [-1.21394365,  0.19687903,  0.02123125, ...,  0.26901965,
         0.31830389,  0.78858745],
       ...,
       [ 1.37386437,  0.33275817,  1.74474449, ...,  2.22423597,
        -1.61212515, -1.48544548],
       [ 1.37386437,  0.20923168,  0.22769377, ...,  1.83492299,
        -1.56825176, -1.40069891],
       [ 1.37386437,  1.39508604,  1.58316512, ...,  1.79166599,
        -1.52437837, -1.42894777]])

In [11]:
y_1= np.transpose(np.array([y])) # Llevar a un vector vertical porque estaba como un vector horizontal
y_1.shape

(178, 1)

In [12]:
y_n= StandardScaler().fit_transform(y_1) # Estandarizacion al vector y
y_n

array([[ 1.01300893],
       [ 0.96524152],
       [ 1.39514818],
       [ 2.33457383],
       [-0.03787401],
       [ 2.23903902],
       [ 1.72952002],
       [ 1.74544249],
       [ 0.94931905],
       [ 0.94931905],
       [ 2.43010864],
       [ 1.69767508],
       [ 1.82505483],
       [ 1.28369089],
       [ 2.54793491],
       [ 1.79320989],
       [ 1.69767508],
       [ 1.22000102],
       [ 2.97147258],
       [ 0.3124203 ],
       [ 0.10542821],
       [ 0.07358327],
       [ 0.91747411],
       [ 0.85378424],
       [ 0.3124203 ],
       [ 0.2646529 ],
       [ 1.42699311],
       [ 1.71359755],
       [ 0.53533487],
       [ 0.91747411],
       [ 1.71359755],
       [ 2.44603111],
       [ 0.7741719 ],
       [ 1.55437286],
       [ 1.10854374],
       [ 0.55125733],
       [ 0.42387759],
       [ 1.14038868],
       [ 0.86970671],
       [ 0.04173834],
       [ 0.15319562],
       [ 0.91747411],
       [ 1.10854374],
       [-0.21302116],
       [ 0.43980005],
       [ 1

In [13]:
print('Shape de X',X_n.shape)
print('Shape de y',y_n.shape)

Shape de X (178, 13)
Shape de y (178, 1)


### **1.2.5 Separamos en entranamiento y test**

In [18]:
X_train, X_test, y_train, y_test = train_test_split(X_n, y_n, test_size=0.25, random_state=42,shuffle=True)

### **1.2.6 Entrenamos el modelo**

In [19]:
regressor = LinearRegression()
regressor.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

### **1.2.7 Analizamos los coeficientes**

In [20]:
#Recuperamos la intersección
print(regressor.intercept_)

[-0.00232517]


In [21]:
print(regressor.coef_)

[[-0.75990723  0.15293898 -0.07397694  0.12222498 -0.08544996  0.08093001
   0.1573841  -0.42133872 -0.04912347  0.07940309  0.36741515  0.03440944
  -0.07374578]]


In [22]:
regressor.coef_[0]

array([-0.75990723,  0.15293898, -0.07397694,  0.12222498, -0.08544996,
        0.08093001,  0.1573841 , -0.42133872, -0.04912347,  0.07940309,
        0.36741515,  0.03440944, -0.07374578])

In [23]:
data_wine.columns

Index(['Wine', 'Alcohol', 'Malic.acid', 'Ash', 'Acl', 'Mg', 'Phenols',
       'Flavanoids', 'Nonflavanoid.phenols', 'Proanth', 'Color.int', 'Hue',
       'OD', 'Proline'],
      dtype='object')

In [24]:
data_wine_x= data_wine.drop(columns=['Proline'])
data_wine_x.columns

Index(['Wine', 'Alcohol', 'Malic.acid', 'Ash', 'Acl', 'Mg', 'Phenols',
       'Flavanoids', 'Nonflavanoid.phenols', 'Proanth', 'Color.int', 'Hue',
       'OD'],
      dtype='object')

### **1.2.5 Calculamos los vectores de media y desviacion estandar**

In [25]:
np.mean(data_wine_x,axis=0)

Wine                     1.938202
Alcohol                 13.000618
Malic.acid               2.336348
Ash                      2.366517
Acl                     19.494944
Mg                      99.741573
Phenols                  2.295112
Flavanoids               2.029270
Nonflavanoid.phenols     0.361854
Proanth                  1.590899
Color.int                5.058090
Hue                      0.957449
OD                       2.611685
dtype: float64

In [109]:
np.std(data_wine_x,axis=0)

Wine                     0.772855
Alcohol                  0.809543
Malic.acid               1.114004
Ash                      0.273572
Acl                      3.330170
Mg                      14.242308
Phenols                  0.624091
Flavanoids               0.996049
Nonflavanoid.phenols     0.124103
Proanth                  0.570749
Color.int                2.311765
Hue                      0.227929
OD                       0.707993
dtype: float64

In [26]:
df=pd.DataFrame(regressor.coef_[0], data_wine_x.columns, columns=['Coeficiente'])
df['Medias']=np.mean(data_wine_x,axis=0)
df['Desviacion']=np.std(data_wine_x,axis=0)
df['Coef_final']=df.Medias + (df.Coeficiente*df.Desviacion) # Calculo del coeficiente sin normalizar
df

Coeficiente     Medias  Desviacion  Coef_final
Wine                    -0.759907   1.938202    0.772855    1.350904
Alcohol                  0.152939  13.000618    0.809543   13.124429
Malic.acid              -0.073977   2.336348    1.114004    2.253938
Ash                      0.122225   2.366517    0.273572    2.399954
Acl                     -0.085450  19.494944    3.330170   19.210381
Mg                       0.080930  99.741573   14.242308  100.894203
Phenols                  0.157384   2.295112    0.624091    2.393334
Flavanoids              -0.421339   2.029270    0.996049    1.609596
Nonflavanoid.phenols    -0.049123   0.361854    0.124103    0.355758
Proanth                  0.079403   1.590899    0.570749    1.636218
Color.int                0.367415   5.058090    2.311765    5.907467
Hue                      0.034409   0.957449    0.227929    0.965292
OD                      -0.073746   2.611685    0.707993    2.559474

### **1.2.5 Calculamos las predicciones**

In [27]:
#Hacemos nuestras predicciones
y_pred = regressor.predict(X_test)
y_pred

array([[ 0.97271604],
       [ 0.94287671],
       [-0.79876967],
       [ 1.11901564],
       [-0.63108268],
       [ 1.37068057],
       [-1.07261686],
       [-0.31064961],
       [-0.71783598],
       [-0.72699756],
       [ 0.63383671],
       [-0.2528659 ],
       [ 0.36343669],
       [-0.59656526],
       [ 1.50225674],
       [-0.91431455],
       [-0.55395307],
       [-0.67027437],
       [ 1.22930137],
       [-0.83556849],
       [ 1.47055375],
       [-0.434413  ],
       [-0.12635708],
       [-0.05434212],
       [-0.1616749 ],
       [-0.06439405],
       [-0.8196852 ],
       [-0.16658075],
       [-0.68067731],
       [ 1.35742301],
       [ 1.06519622],
       [-1.02386761],
       [-0.98795887],
       [ 1.1596794 ],
       [ 1.06193423],
       [ 0.97131572],
       [ 1.0390693 ],
       [-1.01752314],
       [-0.671921  ],
       [ 0.6688603 ],
       [ 0.54171368],
       [-0.72621746],
       [-0.44126773],
       [ 0.12387613],
       [-0.89712194]])

In [28]:
from sklearn import metrics 
print('Mean Absolute Error:', metrics.mean_absolute_error(y_test, y_pred))
print('Mean Squared Error:', metrics.mean_squared_error(y_test, y_pred))
print('Root Mean Squared Error:', np.sqrt(metrics.mean_squared_error(y_test, y_pred)))

Mean Absolute Error: 0.4740948075813741
Mean Squared Error: 0.34272344148046824
Root Mean Squared Error: 0.5854258633511746


In [31]:
media_y= np.mean(data_wine.Proline)
desviacion_y= np.mean(data_wine.Proline)
desviacion_y

746.8932584269663

In [32]:
df_n= pd.DataFrame(X_test, columns=data_wine_x.columns)
df_n['y_pred']=y_pred
df_n['y_pred_final']=(y_pred*desviacion_y)+media_y
df_n

Wine   Alcohol  Malic.acid  ...        OD    y_pred  y_pred_final
0  -1.213944  0.789806    0.685502  ...  1.056952  0.972716   1473.408311
1  -1.213944  1.493907    1.529305  ...  1.014578  0.942877   1451.121518
2   1.373864 -0.087232    0.425180  ... -0.426113 -0.798770    150.297579
3  -1.213944  0.900980   -0.750759  ...  0.138864  1.119016   1582.678492
4   0.079960 -0.778980   -1.046988  ...  1.226445 -0.631083    275.541857
5  -1.213944  1.605081   -0.373741  ...  0.054117  1.370681   1770.645338
6   0.079960 -1.236028    0.981731  ...  0.619094 -1.072617    -54.237045
7   1.373864  0.493343    1.412609  ... -1.485445 -0.310650    514.871156
8   0.079960 -1.717782   -0.885409  ...  0.915707 -0.717836    210.746401
9   1.373864  0.443932    0.200764  ... -0.200123 -0.726998    203.903684
10 -1.213944  0.616869   -0.472483  ...  1.706675  0.633837   1220.301627
11  1.373864  0.616869    0.703455  ... -1.852681 -0.252866    558.029420
12 -1.213944  0.505695    1.349773  ...  0.548472  0.363437   1018.341670
13  0.079960 -0.284874    0.981731  ... -0.694478 -0.596565    301.322691
14 -1.213944  0.777454   -0.472483  ...  0.378979  1.502257   1868.918692
15  0.079960 -0.593691    0.084068  ...  0.237735 -0.914315     63.997888
16  0.079960 -1.964835   -1.432983  ... -0.426113 -0.553953    333.149447
17  0.079960 -1.137207   -1.082894  ...  0.152988 -0.670274    246.269849
18 -1.213944  1.061565   -0.885409  ...  1.325316  1.229301   1665.050161
19  0.079960 -1.137207   -0.849502  ...  0.816836 -0.835568    122.812786
20 -1.213944  1.469202   -0.669969  ...  0.294232  1.470554   1845.239942
21  0.079960  0.135116   -1.190614  ...  0.802712 -0.434413    422.433119
22  0.079960 -0.828391   -1.109824  ... -1.330077 -0.126357    652.518009
23  1.373864  0.493343    2.031997  ... -0.976966 -0.054342    706.305494
24  1.373864 -0.284874    0.048161  ... -1.386574 -0.161675    626.139368
25  1.373864  0.962743    0.380297  ... -1.315952 -0.064394    698.797778
26  0.079960 -0.717217   -0.652016  ...  0.491974 -0.819685    134.675908
27  0.079960 -0.778980   -1.011081  ... -0.440238 -0.166581    622.475220
28  0.079960 -1.137207   -0.454530  ... -0.482611 -0.680677    238.499965
29 -1.213944  0.690985   -0.544297  ...  0.590845  1.357423   1760.743350
30 -1.213944  1.259207   -0.589180  ...  1.381813  1.065196   1542.481138
31  0.079960 -0.778980   -0.634063  ...  0.237735 -1.023868    -17.826554
32  1.373864  0.196879    1.107404  ... -1.315952 -0.987959      8.993436
33 -1.213944  0.715690   -0.607133  ...  0.378979  1.159679   1613.049987
34 -1.213944  0.925685   -0.544297  ...  0.407228  1.061934   1540.044773
35 -1.213944  1.086270   -0.400670  ...  1.339440  0.971316   1472.362421
36  1.373864  1.654492   -0.589180  ... -0.920468  1.039069   1522.967110
37  1.373864 -0.581338    2.848870  ... -1.301828 -1.017523    -13.087917
38  0.079960 -0.778980   -1.136754  ...  0.223610 -0.671921    245.039992
39  1.373864  0.592164   -0.598156  ... -1.174708  0.668860   1246.460506
40 -1.213944  0.085705   -0.750759  ...  0.110615  0.541714   1151.495554
41  0.079960 -0.964270   -0.939268  ...  0.576721 -0.726217    204.486331
42  0.079960 -0.408401   -1.217544  ...  0.774463 -0.441268    417.313362
43  0.079960  0.419227   -1.253450  ... -0.962842  0.123876    839.415502
44  1.373864  0.765101    2.346179  ... -0.793348 -0.897122     76.838933

[45 rows x 15 columns]